### PR TITLE
Update rover wp_bearing per #11284

### PR DESCRIPTION
**Pull request by antholuo**
Attempting to fix https://github.com/ArduPilot/ardupilot/issues/11284

Not sure if I'm doing this correctly since it looks like the TurnRate and GuidedLoiter have been fixed already?

Unsure what the comment on LAT_ACCEL means....

### CODE REVIEW

**Comment by rmackay9** [comment]
Hi @antholuo,

This looks correct to me, thanks!  If possible could you test it in SITL?

The heading-and-speed control isn't used very much so the only way to get the vehicle into this mode is to send a COMMAND_LONG (or COMMAND_INT) with the command field set to MAV_CMD_SET_YAW_SPEED (or something like that).

We've got other commands documented on our [Rover mavlink interface page](https://ardupilot.org/dev/docs/mavlink-rover-commands.html) but not this one.

**Comment by antholuo** [comment]
Hi @rmackay9 !

I'm having a bit of trouble understanding how to send the command (MAV_CMD_SET_YAW_SPEED and MAV_CMD_SET_NAV_YAW_SPEED)?

It's saying that the command is unknown 
image

But the command should be of this format: https://mavlink.io/en/messages/common.html#MAV_CMD_NAV_SET_YAW_SPEED?

**Comment by rmackay9** [comment]
Hi @antholuo,

if you're using mavproxy then I think the easiest way is to use the "long" command.  Another possibility is to type "module load message" and then use the "message" command.  simply typing one of these commands should lead to MAVProxy display a simple "usage" of the command.  Beyond that it could be good to look at examples of how these are used on some of the [AP MAVLink Interface wiki pages](https://ardupilot.org/dev/docs/mavlink-commands.html).

**Comment by antholuo** [comment]
@rmackay9 I ran this through the SITL a few times, and it seems that sending a move command before the previous command has finished will cause the rover to drift quite far from origin: https://youtu.be/yfZ7hb4EYbQ

Doing this on my branch waiting between commands results in a small position error, but doing this on master does not...

Any thoughts on why that might be?
image
Rover far away from origin after multiple successive commands

image
Rover close, but not exactly on origin after multiple commands with wait time between them.

image
Master, which shows no change in position of the rover....

File: Rover/mode_guided.cpp [hunk 1]
```diff
@@ -148,6 +148,7 @@ float ModeGuided::wp_bearing() const
     case Guided_WP:
         return g2.wp_nav.wp_bearing_cd() * 0.01f;
     case Guided_HeadingAndSpeed:
+        return _desired_yaw_cd;
```
Comment (peterbarker) [inline Rover/mode_guided.cpp:151]: This must return centidegrees (see clause above this one)

**Review by peterbarker — CHANGES_REQUESTED** [review]
Your changes here are modifying wp_bearing, where they should be modifying nav_bearing.  We don't have a waypoint to give a bearing to, but we do have a bearing we are trying to acrhieve - so nav_bearing, not wp_bearing

**Pull request closed without merging**